src/api/objects/TransactionManager.js

```diff
@@ -19,31 +19,36 @@
  * this source code distribution or the Licensing information page available
  * at runtime from the About dialog for additional information.
  *****************************************************************************/
-define(['./Transaction'], function (Transaction) {
-    /**
-     * A nested transaction is a transaction which takes place in the context
-     * of a larger parent transaction. It becomes part of the parent
-     * transaction when (and only when) committed.
-     * @param parent
-     * @constructor
-     * @extends {platform/commonUI/edit/services.Transaction}
-     * @memberof platform/commonUI/edit/services
-     */
-    function NestedTransaction(parent) {
-        this.parent = parent;
-        Transaction.call(this, parent.$log);
+
+export default class TransactionManager {
+    constructor() {
+        this.transactions = new Set();
```

Comment:
These are not `transactions` per-se. The transaction is the whole set of object changes. This could be named `dirtyObjects` to make it a little clearer what this is.